src/mlpack/core/cv/meta_info_extractor.hpp

```diff
@@ -0,0 +1,249 @@
+/**
+ * @file meta_info_extractor.hpp
+ * @author Kirill Mishchenko
+ *
+ * Tools for extracting meta information about machine learning algorithms.
+ *
+ * mlpack is free software; you may redistribute it and/or modify it under the
+ * terms of the 3-clause BSD license.  You should have received a copy of the
+ * 3-clause BSD license along with mlpack.  If not, see
+ * http://www.opensource.org/licenses/BSD-3-Clause for more information.
+ */
+#ifndef MLPACK_CORE_CV_META_INFO_EXTRACTOR_HPP
+#define MLPACK_CORE_CV_META_INFO_EXTRACTOR_HPP
+
+#include <type_traits>
+
+#include <mlpack/core.hpp>
+#include <mlpack/core/util/sfinae_utility.hpp>
+
+namespace mlpack {
+namespace cv {
+
+/*
+ * A wrapper struct for holding a Train form.
+ *
+ * @tparam MatType The type of data.
+ * @tparam PredictionsType The type of predictions.
+ * @tparam WeightsType The type of weights.
+ * @tparam DatasetInfo An indicator whether a data::DatasetInfo parameter should
+ *   be present.
+ * @tparam NumClasses An indicator whether the numClasses parameter should be
+ *   present.
+ */
+template<typename MatType,
+         typename PredictionsType,
+         typename WeightsType,
+         bool DatasetInfo,
+         bool NumClasses>
```

Comment:
If I am understanding correctly, you will use this to determine regression or classification types, right?  `NumClasses` should be true for all classification types (some changes still necessary there, would you like me to do them?) and false for all regression types.